### PR TITLE
MGMT-7282: Update operator using subscription channel

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
# Assisted Pull Request

## Description
Instead of patching the catalogsource with the index image, use subscription and update the channel to `alpha` to upgrade the operator from ocm-2.3 to the latest(ocm-2.4)

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested) : Follow the instructions [here](https://github.com/openshift/assisted-service/blob/master/deploy/operator/upgrade/README.md)
- [ ] No tests needed

## Assignees

/cc @djzager 
/cc @lranjbar 
/cc @flaper87 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: deploy/operator/upgrade/upgrade.sh [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
-# wait for olm to update the operator and pods to be recreated
-sleep 15m
+# Upgrade the operator via OLM
+oc patch subscription assisted-service-operator -n assisted-installer --type merge -p "{\"spec\":{\"channel\":\"$CHANNEL\"}}"
 
-echo "## catalogsource spec.image after upgrade"
-oc get catalogsource assisted-service-operator-catalog -n openshift-marketplace -o=jsonpath='{.spec.image}{"\n"}'
+# wait for OLM to update the operator and pods to be recreated
+wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
+wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
```
Comment (djzager) [inline deploy/operator/upgrade/upgrade.sh:17]: Did you mean to wait for both operands?

```suggestion
wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-service"
wait_for_condition "pod" "Ready" "22m" "${ASSISTED_NAMESPACE}"  "app=assisted-image-service"
```

**Review by djzager — COMMENTED** [review]
One comment

File: deploy/operator/upgrade/upgrade.sh (hunk 1 above)
Comment (pawanpinjarkar) [reply deploy/operator/upgrade/upgrade.sh:17]: Yes, What I observed is that after updating the subscription channel, when the operator upgrades, at that time `assisted-service` pod is already in `Ready` state from the previous version but we are sure that the `assisted-image-service` is only created after the upgrade. After the upgrade, `assisted-service` pod is also recreated and so I thought of first checking `assisted-image-service` and then `assisted-service`.

**Review by djzager — APPROVED** [review]

**Review by djzager — APPROVED** [review]

**Pull request merged**